src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,221 @@
+import mate.jdbc.util.ConnectionUtil;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    private static final Logger logger = LogManager.getLogger(CarDaoImpl.class);
+
+    @Override
+    public Car create(Car car) {
+        logger.info("Method create was called with car " + car);
+        String query = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                   PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                car.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create car " + car, e);
+        }
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public Car get(Long id) {
+        logger.info("Method get was called with id " + id);
+        String query = "SELECT c.id AS car_id, model, manufacturer_id, name, country "
+                + "FROM cars AS c "
+                + "JOIN manufacturers AS m ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.is_deleted = false;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCarFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get driver by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return car;
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id AS car_id, model, manufacturer_id, name, country "
+                + "FROM cars AS c "
+                + "JOIN manufacturers AS m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = false;";
+        Car car = null;
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                car = parseCarFromResultSet(resultSet);
+                cars.add(car);
```

Comment:
(suggestion only) Again, I don't think this variable is so important for readability, but it actually depends only on your point of view)